bluetooth: shell: linker error

**Describe the bug**

While trying to build the Bluetooth shell, I faced the following linker error:

```
ction=alignment  -Wl,-u,_OffsetAbsSyms  -Wl,-u,_ConfigAbsSyms  -nostdlib  -static  -Wl,-X  -Wl,-N  -Wl,--orphan-handling=warn && cd /home/gmarull/ws/nordic/zephyrproject/zephyr/build/zephyr && /usr/bin/cmake -E echo
/home/gmarull/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: zephyr/subsys/bluetooth/shell/libsubsys__bluetooth__shell.a(ticker.c.obj): in function `cmd_ticker_info':
/home/gmarull/ws/nordic/zephyrproject/zephyr/subsys/bluetooth/shell/ticker.c:62: undefined reference to `ticker_next_slot_get'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

If the BT shell depends on another module being active (I suspect that this is the case), it should be specified in Kconfig either with a dependency or selection.

**To Reproduce**

```
west build -b nrf52840dk_nrf52840 samples/hello_world -p -- -DCONFIG_BT=y -DCONFIG_BT_SHELL=y
```

Note: I guess I'm missing an option, but it's not easy to know which one since `CONFIG_BT_SHELL` only selects `SHELL`.

**Expected behavior**

I can build the BT shell if `CONFIG_BT_SHELL=y`, or I get a Kconfig error/warning saying what is missing.

**Impact**

Annoyance

**Logs and console output**
See above

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: Zephyr SDK 0.15.0
 - Commit SHA or Version used: 2627063ab21913bf82c5342ce115af1bf18e4299

**Additional context**
N/A

--- a/subsys/bluetooth/shell/Kconfig
+++ b/subsys/bluetooth/shell/Kconfig
@@ -6,6 +6,7 @@
 config BT_SHELL
 	bool "Bluetooth shell"
 	select SHELL
+	select BT_TICKER_NEXT_SLOT_GET if BT_LL_SW_SPLIT
 	help
 	  Activate shell module that provides Bluetooth commands to the
 	  console.

